homeassistant/components/here_weather/config_flow.py

```diff
@@ -0,0 +1,151 @@
+            try:
+                await async_validate_user_input(self.hass, user_input)
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME], data=user_input
+                )
+            except AlreadyConfigured:
+                return self.async_abort(reason="already_configured")
+            except herepy.InvalidRequestError:
+                errors["base"] = "invalid_request"
+            except herepy.UnauthorizedError:
+                errors["base"] = "unauthorized"
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self._get_schema(user_input),
+            errors=errors,
+        )
+
+    def _get_schema(self, user_input: Optional[dict]) -> vol.Schema:
+        known_api_key = None
+        if HERE_API_KEYS in self.hass.data:
+            known_api_key = self.hass.data[HERE_API_KEYS][0]
+        if user_input is not None:
+            return vol.Schema(
+                {
+                    vol.Required(CONF_API_KEY, default=user_input[CONF_API_KEY]): str,
+                    vol.Required(CONF_NAME, default=user_input[CONF_NAME]): str,
+                    vol.Required(CONF_MODE, default=user_input[CONF_MODE]): vol.In(
+                        CONF_MODES
+                    ),
+                    vol.Required(
+                        CONF_LATITUDE, default=user_input[CONF_LATITUDE]
+                    ): cv.latitude,
+                    vol.Required(
+                        CONF_LONGITUDE, default=user_input[CONF_LONGITUDE]
+                    ): cv.longitude,
+                }
+            )
+        return vol.Schema(
+            {
+                vol.Required(CONF_API_KEY, default=known_api_key): str,
+                vol.Required(CONF_NAME, default=DOMAIN): str,
+                vol.Required(CONF_MODE, default=DEFAULT_MODE): vol.In(CONF_MODES),
+                vol.Required(
+                    CONF_LATITUDE, default=self.hass.config.latitude
+                ): cv.latitude,
+                vol.Required(
+                    CONF_LONGITUDE, default=self.hass.config.longitude
+                ): cv.longitude,
+            }
+        )
+
+
+class HereWeatherOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle here_weather options."""
+
+    def __init__(self, config_entry):
+        """Initialize here_weather options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the here_weather options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        options = {
+            vol.Optional(
+                CONF_SCAN_INTERVAL,
+                default=self.config_entry.options.get(
+                    CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+                ),
+            ): int,
+            vol.Optional(CONF_UNIT_SYSTEM, default=self.hass.config.units.name): vol.In(
```

Comment:
Why make this configurable? We should use the HA preference.

Reply:
Done